pgbouncer/datadog_checks/pgbouncer/pgbouncer.py

```diff
@@ -87,51 +95,41 @@ def _collect_stats(self, db, instance_tags):
-
-                metrics = scope['metrics']
-                cols = [m[0] for m in metrics]
-
-                try:
+            with db.cursor(cursor_factory=pgextras.DictCursor) as cursor:
+                for scope in metric_scope:
+                    descriptors = scope['descriptors']
+                    metrics = scope['metrics']
                     query = scope['query']
-                    self.log.debug("Running query: %s" % query)
-                    cursor.execute(query)
 
-                    results = cursor.fetchall()
+                    try:
+                        self.log.debug("Running query: %s", query)
+                        cursor.execute(query)
 
-                except pg.Error as e:
-                    self.log.warning("Not all metrics may be available: %s" % str(e))
-                    continue
+                        rows = cursor.fetchall()
 
-                for row in results:
-                    if row[0] == self.DB_NAME:
-                        continue
+                    except pg.Error as e:
+                        self.log.exception("Not all metrics may be available")
 
-                    desc = scope['descriptors']
+                    else:
+                        for row in rows:
+                            self.log.debug("Processing row: %r", row)
 
-                    # Some versions of pgbouncer have extra fields at the end of SHOW POOLS
-                    if len(row) == len(cols) + len(desc) + 1:
-                        row = row[:-1]
-                    elif len(row) == len(cols) + len(desc) + 2:
-                        row = row[:-2]
+                            # Skip the "pgbouncer" database
+                            if row['database'] == self.DB_NAME:
+                                continue
 
-                    assert len(row) == len(cols) + len(desc)
+                            tags = list(instance_tags)
+                            tags += ["%s:%s" % (tag, row[column]) for (column, tag) in descriptors]
+                            for (column, (name, reporter)) in metrics:
+                                if column in row:
+                                    reporter(self, name, row[column], tags)
 
-                    tags = list(instance_tags)
-                    tags += ["%s:%s" % (d[0][1], d[1]) for d in zip(desc, row[:len(desc)])]
-                    for i, (key_name, (mname, mtype)) in enumerate(metrics):
-                        value = row[i + len(desc)]
-                        mtype(self, mname, value, tags)
+                        if not rows:
+                            self.log.warning("No results were found for query: %s", query)
 
-            if not results:
-                self.warning('No results were found for query: "%s"' % query)
-
-            cursor.close()
         except pg.Error as e:
-            self.log.error("Connection error: %s" % str(e))
+            self.log.exception("Connection error")
```

Comment:
Also `e` is never used which make the `flake8` fail (that's why the travis tests are red).

Reply:
I'll remove!